src/libstd/sys/unix/ext/fs.rs

```diff
@@ -215,36 +215,312 @@ impl OpenOptionsExt for OpenOptions {
 // casts and rely on manual lowering to `stat` if the raw type is desired.
 #[stable(feature = "metadata_ext", since = "1.1.0")]
 pub trait MetadataExt {
+    /// Returns the ID of the device containing the file.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(target_os = "linux")] {
+    /// use std::fs;
+    /// use std::os::unix::prelude::MetadataExt;
+    ///
+    /// # use std::io;
+    /// # fn f() -> io::Result<()> {
+    /// let meta = fs::metadata("some_file")?;
+    /// let dev_id = meta.dev();
+    /// # Ok(())
+    /// # }
+    /// # }
+    /// ```
     #[stable(feature = "metadata_ext", since = "1.1.0")]
     fn dev(&self) -> u64;
+    /// Returns the inode number.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(target_os = "linux")] {
+    /// use std::fs;
+    /// use std::os::unix::prelude::MetadataExt;
+    ///
+    /// # use std::io;
+    /// # fn f() -> io::Result<()> {
+    /// let meta = fs::metadata("some_file")?;
+    /// let inode = meta.ino();
+    /// # Ok(())
+    /// # }
+    /// # }
+    /// ```
     #[stable(feature = "metadata_ext", since = "1.1.0")]
     fn ino(&self) -> u64;
+    /// Returns the rights applied to this file.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(target_os = "linux")] {
+    /// use std::fs;
+    /// use std::os::unix::prelude::MetadataExt;
+    ///
+    /// # use std::io;
+    /// # fn f() -> io::Result<()> {
+    /// let meta = fs::metadata("some_file")?;
+    /// let mode = meta.mode();
+    /// let user_has_write_access      = mode & 0200;
+    /// let user_has_read_write_access = mode & 0600;
+    /// let group_has_read_access      = mode & 0040;
+    /// let others_have_exec_access    = mode & 0001;
```

Comment:
Pretty sure these should all be `0o200` etc? I didn't think rust had zero-prefix octal notation.

Reply:
Oh damn you're right! Bad assumption. :s